app/models/book.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Book
+class Book < ApplicationRecord
+  FILTERS = %i[newest popular price_asc price_desc
+               by_title_asc by_title_desc].freeze
+  DEFAULT_FILTER = :newest
+
+  has_many :images, dependent: :destroy
+  belongs_to :category
+  has_and_belongs_to_many :authors
+  has_many :positions
+  has_many :orders, through: :positions, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  validates :title, :price, :materials, presence: true
+  validates :height, :width, :depth,
+            numericality: { greater_than_or_equal_to: 0 }
+  validates :publication_year,
+            numericality: { less_than_or_equal_to: Time.current.year }
+
+  scope :for_slider, -> { order(:created_at).last(3) }
+  scope :best_sellers, -> { where(best_seller: true).last(4) }
+  scope :newest, -> { order('created_at DESC') }
+  scope :popular, -> { where(popular: true) }
```

Comment:
I think you misunderstood how the popular scope should work. The most popular book is the book that was ordered more time than the others.
For example:
http://joxi.ru/823pLJKSJoJ19A
I ordered 2 Death Be Not Proud books and 1 Cabbages and Kings book. So, Death Be Not Proud  should be more popular (if there were no more orders).

Reply:
Please help me, I need to get hold of books, but I do not know how to build a query.

Reply:
I think checking if the order was delivered is unnecessary in this case. What query are you using?

Reply:
What's my mistake ?
![2018-07-20 13 39 48](https://user-images.githubusercontent.com/9571192/42998325-79463040-8c22-11e8-88d2-22e675b6a541.png)
Probably not every Book has Positions, and not have a quantity

Reply:
Awesome! Thank you!